pkg/reconciler/v1alpha1/taskrun/resources/pod.go

```diff
@@ -155,6 +161,39 @@ func makeCredentialInitializer(serviceAccountName, namespace string, kubeclient
 	}, volumes, nil
 }
 
+func makeWorkingDirInitializer(taskSpec v1alpha1.TaskSpec) *corev1.Container {
```

Comment:
I'd really like to see this as a publicly exported function with its own unit tests, and i think we can tighten up the interface a bit more: e.g. pass in just the `steps` instead of the entire task spec (or do the work to extract the workingDirs from the `steps` before calling this function?)

then if we had that we could take or leave the additional cases in `TestMakePod` which i think are a bit more verbose than we'd see in some slightly more targeted unit tests